src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -63,14 +64,30 @@ internal SdkDirectoryWorkloadManifestProvider(string sdkRootPath, string sdkVers
             }
 
             _sdkRootPath = sdkRootPath;
+            _sdkOrUserLocalPath = sdkRootPath;
             _sdkVersionBand = new SdkFeatureBand(sdkVersion);
             _workloadSetVersionFromConstructor = workloadSetVersion;
             _globalJsonPathFromConstructor = globalJsonPath;
 
-            var knownManifestIdsFilePath = Path.Combine(_sdkRootPath, "sdk", sdkVersion, "KnownWorkloadManifests.txt");
+            string? userManifestsRoot = userProfileDir is null ? null : Path.Combine(userProfileDir, "sdk-manifests");
+            string dotnetManifestRoot = Path.Combine(_sdkOrUserLocalPath, "sdk-manifests");
+            if (userManifestsRoot != null && WorkloadFileBasedInstall.IsUserLocal(_sdkRootPath, _sdkVersionBand.ToString()) && Directory.Exists(userManifestsRoot))
+            {
+                _sdkOrUserLocalPath = userProfileDir ?? _sdkRootPath;
+                if (getEnvironmentVariable(EnvironmentVariableNames.WORKLOAD_MANIFEST_IGNORE_DEFAULT_ROOTS) == null)
+                {
+                    _manifestRoots = new[] { userManifestsRoot, dotnetManifestRoot };
+                }
+            }
+            else if (getEnvironmentVariable(EnvironmentVariableNames.WORKLOAD_MANIFEST_IGNORE_DEFAULT_ROOTS) == null)
+            {
+                _manifestRoots = new[] { dotnetManifestRoot };
+            }
+
+            var knownManifestIdsFilePath = Path.Combine(_sdkOrUserLocalPath, "sdk", sdkVersion, "KnownWorkloadManifests.txt");
             if (!File.Exists(knownManifestIdsFilePath))
             {
-                knownManifestIdsFilePath = Path.Combine(_sdkRootPath, "sdk", sdkVersion, "IncludedWorkloadManifests.txt");
+                knownManifestIdsFilePath = Path.Combine(_sdkOrUserLocalPath, "sdk", sdkVersion, "IncludedWorkloadManifests.txt");
```

Comment:
I think that `KnownWorkloadManifests.txt` and `IncludedWorkloadManifests.txt` are files that are part of the SDK installation, and would not be expected to be in the user local path.  So I think we should probably still use `_sdkRootPath` when constructing these paths.  Does that make sense?

Reply:
I didn't know anything about them, so I thought they might be writable, in which case they should be in the user local folder, but if they don't go there, I can move it back to _sdkRootPath.